library/src/main/scala/scala/scalajs/js/typedarray/Float32Array.scala

```diff
@@ -18,7 +18,7 @@ class Float32Array private extends TypedArray[Float, Float32Array] {
   def this(typedArray: TypedArray[_, _]) = this()
 
   /** Creates a new Float32Array with the elements in the given array */
-  def this(array: js.Array[_]) = this()
+  def this(array: js.Iterable[_]) = this()
```

Comment:
Spec says there is one constructor taking `object` that forwards to `TypedArray.from`, which in turn dynamically checks whether `@@iterator` is defined. If it is, elements are extracted using an iterator. Otherwise, it is assumed `object` is an array-like object and extraction is performed using indexed access.

Given that `js.Array[T] <: js.Iterable[T]`, this seemed the most reasonable typing, although indeed, the constructor taking a `TypedArray` also should go away. This seems like the most reasonable option to me.